### PR TITLE
use MachinePoolStatusFailure on machinePool

### LINKED ISSUE #3825: Change type of `FailureReason` for `AzureMachinePoolStatus` from `MachineStatusError` to `MachinePoolStatusFailure`
/kind bug

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
In the status field for AzureMachinePool, the failure reason is of type `FailureReason *errors.MachineStatusError`. However, there is another type defined called `errors.MachinePoolStatusFailure` that would be a better fit.

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by bravebeaver**
**What type of PR is this?**
/kind bug 

change type of FailureReason on AzureMachinePoolStatus

**What this PR does / why we need it**:
- set failure reason with MachinePoolStatusFailure type

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3825 

**Special notes for your reviewer**:

Change types on AzureMachinePoolStatus type on the API, which might have external impact.

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [X] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by Jont828 — COMMENTED** [review]
Looks good so far - just a comment and I think you got a test failure.

File: exp/controllers/azuremachinepool_controller_test.go [hunk 1]
```diff
@@ -97,27 +142,46 @@ func TestAzureMachinePoolReconcilePaused(t *testing.T) {
 			},
 		},
 	}
-	g.Expect(c.Create(ctx, cluster)).To(Succeed())
+}
 
-	azCluster := &infrav1.AzureCluster{
+func getMachinePool(name, namespace, secretName string) *expv1.MachinePool {
```
[the remark this answers is not in the document]
Comment (bravebeaver) [reply exp/controllers/azuremachinepool_controller_test.go]: sorry it just took me so long to realise i missed github notifications 😅,. updated as suggested

**Comment by dtzar** [comment]
@bravebeaver - I think you just need to fix the e2e test(s) and then you should be good

**Pull request closed without merging**